lib/matplotlib/axes/_axes.py

```diff
@@ -5055,11 +5055,9 @@ def fill(self, *args, data=None, **kwargs):
         """
         # For compatibility(!), get aliases from Line2D rather than Patch.
         kwargs = cbook.normalize_kwargs(kwargs, mlines.Line2D)
-
-        patches = []
-        for poly in self._get_patches_for_fill(*args, data=data, **kwargs):
+        patches = [*self._get_patches_for_fill(*args, data=data, **kwargs)]
```

Comment:
Did we need the repack for this?

Reply:
yes, get_patches_for_fill is a generator but we want a list

Reply:
I had the same question and dug up the `_get_patches_for_fill` implementation to check that this really necessary.

Maybe in the future, add a comment in such cases, so that not everybody is wondering why this is there.

Reply:
Fair point; I smuggled a comment as an additional commit in https://github.com/matplotlib/matplotlib/pull/14695.

Reply:
Smart move, using the comment as a reminder on a two week-old PR. Merged :rofl:

Reply:
Well, I could have opened another one-line PR if you prefer :p